src/components/color-picker/color-field/color-field.tsx

```diff
@@ -0,0 +1,494 @@
+import { SyncWithRAF } from "../../../common/sync-with-frames";
+import { ColorFormat, ColorVariants } from "../../../common/types";
+import { getColorFormat } from "../utils/color-format";
+import { fromStringToColorVariants } from "../utils/color-variants";
+import { ColorFieldTranslations } from "./translations";
+
+type KeyEvents =
+  | typeof KEY_CODES.ARROW_UP
+  | typeof KEY_CODES.ARROW_RIGHT
+  | typeof KEY_CODES.ARROW_DOWN
+  | typeof KEY_CODES.ARROW_LEFT;
+
+@Component({
+  formAssociated: true,
+  shadow: { delegatesFocus: true },
+  styleUrl: "color-field.scss",
+  tag: "ch-color-field"
+})
+export class ChColorField {
+  #canvasRef: HTMLCanvasElement;
+  #colorPickedVariants: ColorVariants;
+  #colorFormat: ColorFormat = "hex";
+  #currentX: number = 0;
+  #currentY: number = 0;
+  #dragRAF: SyncWithRAF;
+  #isDragging: boolean = false;
+  #markerRef: HTMLDivElement;
+  #resizeObserver: ResizeObserver;
+  #resizeRAF: SyncWithRAF = new SyncWithRAF();
+  #isUserAction: boolean = false;
+
+  #keyEvents: {
+    [key in KeyEvents]: (event: KeyboardEvent) => void;
+  } = {
+    [KEY_CODES.ARROW_UP]: ev => {
+      this.#currentY -= this.step;
+      if (this.#currentY < 0) {
+        this.#currentY = 0;
+      }
+      this.#drawMarker();
+      this.#emitColorChange();
+      ev.preventDefault();
+    },
+
+    [KEY_CODES.ARROW_DOWN]: ev => {
+      this.#currentY += this.step;
+      const canvasHeight = this.#canvasRef?.height || 0;
+      if (this.#currentY >= canvasHeight) {
+        this.#currentY = canvasHeight;
+      }
+      this.#drawMarker();
+      this.#emitColorChange();
+      ev.preventDefault();
+    },
+
+    [KEY_CODES.ARROW_LEFT]: ev => {
+      this.#currentX -= this.step;
+      if (this.#currentX < 0) {
+        this.#currentX = 0;
+      }
+      this.#drawMarker();
+      this.#emitColorChange();
+      ev.preventDefault();
+    },
+
+    [KEY_CODES.ARROW_RIGHT]: ev => {
+      this.#currentX += this.step;
+      const canvasWidth = this.#canvasRef?.width || 0;
+      if (this.#currentX >= canvasWidth) {
+        this.#currentX = canvasWidth;
+      }
+      this.#drawMarker();
+      this.#emitColorChange();
+      ev.preventDefault();
+    }
+  };
+
+  @AttachInternals() internals: ElementInternals;
+
+  @Element() el: HTMLChColorFieldElement;
+
+  /**
+   * Internal state to store the color variants.
+   */
+  @State() private colorVariants: ColorVariants;
+
+  /**
+   * Step to navigate on the canvas.
+   */
+  @Prop() readonly step: number = 1;
+
+  /**
+   * Specifies if the color field is disabled.
+   */
+  @Prop() readonly disabled: boolean = false;
+
+  /**
+   * Specifies if the color field is readonly.
+   */
+  @Prop() readonly readonly: boolean = false;
+
+  /**
+   * Specifies the literals required in the control.
+   */
+  @Prop() readonly translations: ColorFieldTranslations = {
+    accessibleName: {
+      description: "2D color selector",
+      label: "Color field"
+    }
+  };
```

Comment:
In all components, we manage the accessible literals as separate properties.

```suggestion
  /**
   * Specifies the literals required in the control.
   */
  @Prop() readonly accessibleName: string = "Color field";
  
  @Prop() readonly accessibleRoleDescription: string = "2D color field";
```